Topics/Development_Process/MongoDB_in_Docker.md

```diff
@@ -0,0 +1,110 @@
+
+```console
+docker pull mongodb/mongodb-community-server
+```
+
+Alternatively you can also find MongoDB's and other images through Docker Desktop's search bar. 
+
+If you just want to get a database server up and running, the easiest way is to use Docker Desktop's GUI. Navigate to the 'Images' tab on the left hand tab, and find the 'mongodb-community-server' image that you just pulled. Under actions click the run button and it will open up a popup prompting you to input optional settings. The only absolutely necessary field to input to get up and running is the 'Ports' field. [Read about ports here.](./MongoDB_in_Docker.md#a-quick-summary-about-ports) 
+
+In the port field, you can see that Docker Desktop already fills in the container port and control protocol for you (`27017/tcp`), so all we have to do is input the host port. We can use the same port (`27017`) as common.
+
+If you have multiple MongoDB containers running at once, using `27016` or other numbers around is also possible. 
+
+If you have other images running as well, docker will not allow multiple images to use the same port.
+
+Now that you have gotten your first MongoDB container up and running, lets explore the other optional settings.
+
+- Container name is self explanatory, this makes it easier to find and manage running containers
+- Volumes allows for data to persist even when the container is removed/not running. 
+    - This is often ideal when having to make changes to the image and recomposing, as you wouldn't want to lose your databases' data.
+    - In MongoDB's case, the container path **MUST** be '/data/db'
+        - This corresponds to which folder in the container you want to be externally mounted
+    - The Host path can then be any folder you like. This is where the data will live in your local machine. 
+        - If you want to use this data in an another MongoDB container, you can simply remount the host path to the same container path.
+    - The environment variables for MongoDB are generally the root user and password.
+        - MONGO_INITDB_ROOT_USERNAME, MONGO_INITDB_ROOT_PASSWORD are the variables
+        - You can set the value of these to be whatever you want
+
+In a final summary, what we have done so far is setup a MongoDB server in a docker container. Following the steps in the NoSQL page about MongoDB, using MongoDB compass, you can access your database through `'localhost:27017'` and putting username and password that you have set using the environment variables.
+
+In one command line, you can use:
+
+```console
+docker run --name containername -p 27017:27017 -e MONGO_INITDB_ROOT_USERNAME=user -e MONGO_INITDB_ROOT_PASSWORD=pass mongodb/mongodb-community-server:latest
+```
+
+From what we discussed above, you can infer that -p is publish port, -e is an environment variable, and the final argument is the image. MongoDB recommends specifying the version instead of latest to ensure the version used is the same as expected. 

```

Comment:
Possibly bold publish port, environmental variable, and image as they are variables that will be in use.

Reply:
That is a good idea. Since I was familiar with the topic I figured the flags would be easy to recognize, but highlighting them would help. However for the image since it is an argument and not a flag I figure that I should just clarify more as to what is is since highlighting flags as well as argument descriptions would confuse the reader.